src/overlays/actors/ovl_En_Trt2/z_en_trt2.c

```diff
@@ -76,111 +103,832 @@ static DamageTable D_80AD58D8 = {
+    Math_ApproachF(&this->actor.velocity.y, 1.5f, 0.2f, 0.1f);
+
+    if (this->actor.world.pos.y > 200.0f) {
+        if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+            this->unk_3B2 = 0x13;
+            ActorCutscene_Stop(this->unk_3DA);
+        } else {
+            s32 i;
+            Vec3f sp68;
+            Vec3f sp5C = { 0.0f, 0.5f, 0.0f };
+
+            for (i = (s32)this->skelAnime.animLength - (s32)this->skelAnime.curFrame; i >= 0; i -= 2) {
+                sp68.x = randPlusMinusPoint5Scaled(60.0f) + this->actor.world.pos.x;
+                sp68.z = randPlusMinusPoint5Scaled(60.0f) + this->actor.world.pos.z;
+                sp68.y = randPlusMinusPoint5Scaled(50.0f) + (this->actor.world.pos.y + 20.0f);
+                func_800B3030(globalCtx, &sp68, &sp5C, &sp5C, 100, 0, 3);
+            }
+
+            func_800B9010(&this->actor, NA_SE_EN_COMMON_EXTINCT_LEV - SFX_FLAG);
+        }
+    } else if (this->actor.world.pos.y < 5.0f) {
+        func_80AD4A78(this, globalCtx);
+    }
+
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+
+    if ((this->actor.shape.rot.y >= 0x2800) && (this->actor.shape.rot.y < 0x3800)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_KOTAKE_ROLL);
+    }
+}
+
+void func_80AD4550(EnTrt2* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    u8 sp23 = func_80152498(&globalCtx->msgCtx);
+
+    if ((player->transformation != PLAYER_FORM_HUMAN) && (player->transformation != PLAYER_FORM_FIERCE_DEITY)) {
+        func_80AD3380(&this->skelAnime, sAnimations, 7);
+        this->unk_3B2 = 17;
+    }
+
+    if ((sp23 == 5) && func_80147624(globalCtx)) {
+        globalCtx->msgCtx.unk11F22 = 0x43;
+        globalCtx->msgCtx.unk12023 = 4;
+    }
+}
+
+void func_80AD4608(EnTrt2* this) {
+    s16 temp = this->unk_3B6 - 1;
+
+    if (temp != 0) {
+        this->unk_3B6 = temp;
+        return;
+    }
+
+    temp = this->unk_3B8 + 1;
+    if (temp >= 3) {
+        this->unk_3B8 = 0;
+        this->unk_3B6 = (s32)(Rand_ZeroOne() * 60.0f) + 20;
+    } else {
+        this->unk_3B8 = temp;
+        this->unk_3B6 = 1;
+    }
+}
+
+void func_80AD469C(EnTrt2* this, GlobalContext* globalCtx) {
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_trt_Skel_00FEF0, &object_trt_Anim_000A44, NULL, NULL, 0);
+    this->actor.draw = func_80AD56E8;
+}
+
+void func_80AD46F8(EnTrt2* this) {
+    if ((this->actor.home.pos.y + 60.0f) < this->actor.world.pos.y) {
+        this->actor.velocity.y = -0.2f;
+    } else if (this->actor.world.pos.y <= (this->actor.home.pos.y + 50.0f)) {
+        this->actor.velocity.y = 0.2f;
+    }
+}
+
+s32 func_80AD475C(EnTrt2* this, Path* path, s32 arg2) {
+    Vec3s* points;
+    s32 count;
+    f32 phi_f12;
+    s32 ret;
+    f32 phi_f14;
+    s32 arg = arg2;
+    f32 sp44;
+    f32 sp40;
+    f32 sp3C;
+    Vec3f sp30;
+
+    points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
```

Comment:
The cast isn't necessary here
```suggestion
    points = Lib_SegmentedToVirtual(path->points);
```